code/_onclick/hud/ability_screen_objects.dm

```diff
@@ -60,9 +112,28 @@
 		overlays.Add(open_state)
 	update_icon()
 
-/obj/screen/movable/ability_master/proc/open_ability_master()
+/obj/screen/ability_master/proc/open_ability_master()
+	var/list/screen_loc_xy = splittext(screen_loc,",")
+
+	//Create list of X offsets
+	var/list/screen_loc_X = splittext(screen_loc_xy[1],":")
+	var/x_position = decode_screen_X(screen_loc_X[1], my_mob)
```

Comment:
I feel like somewhere, maybe here or maybe in the decoder proc, you need to check to make sure this is a legitimate x position in the sense that when it's re-encoded you get something sane. Maybe it already does the fallback correctly and I'm paranoid, maybe it already fields everything it should and so if this is `null` or whatever you should crash; I'm not sure.

Reply:
I think I am going to sit down and implement a set of helpers for encoding and deencoding arbitrary screen locs so I can use them in other UI stuff beyond this, as well as stuff like the click catcher horseshit we were discussing the other day. This code is stuff I've restored from commit history rather than something I wrote, for reference. I did test the PR overall but I have no confidence that this is the best way or even an appropriate way to handle screen locs.

Reply:
I think that's a good idea. With the way the client screen is set up currently, we'll be seeing more and more need to mess with client-dependent screen locations.